src/dmd/expressionsem.d

```diff
@@ -5282,14 +5283,38 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
     private Expression compileIt(CompileExp exp)
     {
-        //printf("CompileExp::compileIt('%s')\n", exp.toChars());
-        auto se = semanticString(sc, exp.e1, "argument to mixin");
-        if (!se)
-            return null;
-        se = se.toUTF8(sc);
+        OutBuffer buf;
+        if (exp.exps)
+        {
+            foreach (ex; *exp.exps)
+            {
+                sc = sc.startCTFE();
+                auto e = ex.expressionSemantic(sc);
+                e = resolveProperties(sc, e);
+                sc = sc.endCTFE();
+
+                // allowed to contain types as well as expressions
+                e = ctfeInterpretForPragmaMsg(e);
```

Comment:
This should be tested. It also makes it clear as to what you mean by "like `pragma(msg,..);`"

Reply:
As a later refactor, I intend to have `pragma(msg)` and `mixin(args)` share the same implementation, which will minimize needed tests and ensure they behave the same.